src/stores/ui.ts

```diff
@@ -1,33 +1,148 @@
+   * the filters should be closed on SSR for the first load on desktop layout.
+   */
+  isFilterDismissed: boolean
+  /**
+   * whether the site layout is desktop (or mobile).
+   */
+  isDesktopLayout: boolean
+  /**
+   * whether the request user agent is mobile or not.
+   */
+  isMobileUa: boolean
 }
 
 export const useUiStore = defineStore('ui', {
   state: (): UiState => ({
-    instructionsSnackbarState: 'not_shown',
+    instructionsSnackbarState: NOT_SHOWN,
+    isFilterVisible: false,
+    isFilterDismissed: false,
+    isDesktopLayout: false,
+    isMobileUa: true,
   }),
 
   getters: {
-    areInstructionsVisible(state) {
-      return state.instructionsSnackbarState === 'visible'
+    areInstructionsVisible(state): boolean {
+      return state.instructionsSnackbarState === VISIBLE
     },
   },
 
   actions: {
     showInstructionsSnackbar() {
-      if (this.instructionsSnackbarState === 'not_shown') {
-        this.instructionsSnackbarState = 'visible'
+      if (this.instructionsSnackbarState === NOT_SHOWN) {
+        this.instructionsSnackbarState = VISIBLE
       }
     },
     hideInstructionsSnackbar() {
-      this.instructionsSnackbarState = 'dismissed'
+      this.instructionsSnackbarState = DISMISSED
+    },
+    /**
+     * Given a list of key value pairs of UI state parameters and their states,
+     * populate the store state to match the cookie.
+     *
+     * Values stored in the cookie are stored across sessions and are modified
+     * if the breakpoints change.
+     *
+     * @param cookies - mapping of UI state parameters and their states.
+     */
+    initFromCookies(cookies: UiStateCookie) {
+      this.isDesktopLayout = cookies.isDesktopLayout ?? false
+      this.isMobileUa = cookies.isMobileUa ?? false
+      this.isFilterDismissed = cookies.isFilterDismissed ?? false
+      this.isFilterVisible = this.isDesktopLayout
+        ? !this.isFilterDismissed
+        : false
+    },
+    /**
+     * Sets the breakpoint and ua state. We have to set the cookie value inside
+     * the `setup` function because `app.$cookies` is not available outside it.
+     * This is why we return the object that needs to be saved in the cookies.
+     * @param isDesktopLayout - whether the layout is desktop (`lg` with the `new_header`
+     * and `md` with the `old_header`).
+     * @param isMobileUa - whether the UA is `mobile` or not.
+     */
+    updateBreakpoint(
+      isDesktopLayout: boolean,
+      isMobileUa: boolean | undefined
+    ): ScreenParameters | undefined {
+      if (
+        this.isDesktopLayout === isDesktopLayout &&
+        typeof isMobileUa === 'undefined'
+      )
+        return
+      this.isDesktopLayout = isDesktopLayout
+      // Only update `isMobileUa` if the value is provided
+      if (typeof isMobileUa !== 'undefined') {
+        this.isMobileUa = isMobileUa
+      }
+      return {
+        isDesktopLayout: this.isDesktopLayout,
+        isMobileUa: this.isMobileUa,
+      }
```

Comment:
Would just like to add some whitespace and additional curlies here to clarify what is happening. The naked conditional `return` was confusing to read :confounded: 

```suggestion
      if (
        this.isDesktopLayout === isDesktopLayout &&
        typeof isMobileUa === 'undefined'
      ) {
        return
      }

      this.isDesktopLayout = isDesktopLayout

      // Only update `isMobileUa` if the value is provided
      if (typeof isMobileUa !== 'undefined') {
        this.isMobileUa = isMobileUa
      }
 
      return {
        isDesktopLayout: this.isDesktopLayout,
        isMobileUa: this.isMobileUa,
      }
```

Reply:
I've updated the `isMobileUa` handling, cookie handling, and the structure of this method as a whole. Thank you for giving ideas and pushing me to improve it all!